libc/string_tokpbrkstr.c

```diff
@@ -0,0 +1,633 @@
+static void create_ascii_set(signed int x)
+{
+	int i;
+
+	for (i = x; i < ASCII_LENGTH; i++) {
+		asciiStr[i - x] = i;
+	}
+}
+
+
+static void create_extAscii_set(void)
+{
+	int i;
+
+	for (i = 1; i < EXTENDED_ASCII_LENGTH; i++) {
+		extAsciiStr[i - 1] = i;
+	}
+}
+
+
+TEST_SETUP(string_tok)
+{
+}
+
+
+TEST_TEAR_DOWN(string_tok)
+{
+}
+
+
+TEST(string_tok, basic)
+{
+	char *holder, dividers[4];
+
+	/* Create an Ascii array that starts from 1*/
+	create_ascii_set(1);
+
+	sprintf(dividers, "%c%c%c", asciiStr[1], asciiStr[sizeof(asciiStr) / 2], asciiStr[sizeof(asciiStr) - 1]);
+
+	/* Checking if the return of strtok is a pointer equal to the input string pointer in the first tokenization */
+	TEST_ASSERT_EQUAL_PTR(asciiStr, strtok(asciiStr, &asciiStr[sizeof(asciiStr) - 1]));
+
+	/* Holder is used to capture the actual state of tokenization and assert its value to expected sets */
+	holder = strtok(asciiStr, dividers);
+	TEST_ASSERT_EQUAL_PTR(asciiStr, holder);
+	/*
+	 * In this case strtok cut of token which have length of 2 where the first element is saved as
+	 * value in the holder but the second is a divider and was discarded
+	 */
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(asciiStr, holder, sizeof(holder));
+
+	/* Using NULL as input determinate to use same input but without the last tokenization */
+	holder = strtok(NULL, dividers);
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[2], holder);
+	/*
+	 * Starting from 2 because of the last strtok cut of the first two elements for token
+	 * contains output and divider
+	 */
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(&asciiStr[2], holder, sizeof(holder));
+
+	holder = strtok(NULL, dividers);
+	/* Adding here 1 because always is taken next place after the point of divide */
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[sizeof(asciiStr) / 2 + 1], holder);
+	/*
+	 * Starting from 2 because of the last strtok cut of the first two elements for token
+	 * contains output and divider
+	 */
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(&asciiStr[sizeof(asciiStr) / 2 + 1], holder, sizeof(holder));
+
+	holder = strtok(NULL, dividers);
+	/*
+	 * After getting everything available from the string
+	 * pointer will be set as 0 and the value of the next reads will be NULL
+	 */
+	TEST_ASSERT_EQUAL_PTR(NULL, holder);
+	TEST_ASSERT_NULL(holder);
+}
+
+TEST(string_tok, null_char_impact)
+{
+	char holder[ASCII_LENGTH];
+
+	create_ascii_set(1);
+	asciiStr[ASCII_LENGTH / 2] = '\0';
+
+	/* Last element of ascii string is \127 and between 0 and 127 was written 0*/
+	memcpy(holder, strtok(asciiStr, "\127"), ASCII_LENGTH);
+	create_ascii_set(1);
+
+	/* \0 was placed in half of the ASCII set. This case is looking for sensitivity for \0 in strings*/
+	TEST_ASSERT_EQUAL_CHAR_ARRAY(holder, asciiStr, ASCII_LENGTH / 2);
+	TEST_ASSERT_NOT_EQUAL_INT(0, memcmp(holder, asciiStr, ASCII_LENGTH));
+}
+
+
+TEST(string_tok, empty_input_output)
+{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wrestrict"
+
```

Comment:
Describe why this pragma is needed. You can also do it in different way to not use pragma, but I don't think it's worth to do it.